indexer/migrations/20230523_create_schema.sql

```diff
@@ -0,0 +1,99 @@
+/**
+ * BLOCK DATA
+ */
+
+CREATE TABLE IF NOT EXISTS l1_block_headers (
+	hash        VARCHAR NOT NULL PRIMARY KEY,
+	parent_hash VARCHAR NOT NULL,
+	number      NUMERIC NOT NULL,
+	timestamp   INTEGER NOT NULL,
+
+    CONSTRAINT
+        l1_parent_hash_fkey FOREIGN KEY(parent_hash) REFERENCES l1_block_headers(hash)
+);
+
+CREATE TABLE IF NOT EXISTS l2_block_headers (
+    -- Block header
+	hash                     VARCHAR NOT NULL PRIMARY KEY,
+	parent_hash              VARCHAR NOT NULL,
+	number                   NUMERIC NOT NULL,
+	timestamp                INTEGER NOT NULL,
+
+    -- Finalization information
+    l1_block_hash            VARCHAR NOT NULL REFERENCES l1_block_headers(hash),
+    legacy_state_batch_index INTEGER,
+
+    CONSTRAINT
+        l2_parent_hash_fkey FOREIGN KEY(parent_hash) REFERENCES l2_block_headers(hash)
+);
+
+CREATE TABLE IF NOT EXISTS legacy_state_batches (
+	index         INTEGER NOT NULL PRIMARY KEY,
+	root          VARCHAR NOT NULL,
+	size          INTEGER NOT NULL,
+	prev_total    INTEGER NOT NULL,
+
+    -- Finalization information
+	l1_block_hash VARCHAR NOT NULL REFERENCES l1_block_headers(hash)
+);
+
+/** 
+ * EVENT DATA
+ */
+
+CREATE TABLE IF NOT EXISTS l1_contract_events (
+    guid             VARCHAR NOT NULL PRIMARY KEY,
+	block_hash       VARCHAR NOT NULL REFERENCES l1_block_headers(hash),
+    transaction_hash VARCHAR NOT NULL,
+    event_signature  VARCHAR NOT NULL,
+    log_index        INTEGER NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS l2_contract_events (
+    guid             VARCHAR NOT NULL PRIMARY KEY,
+	block_hash       VARCHAR NOT NULL REFERENCES l2_block_headers(hash),
+    transaction_hash VARCHAR NOT NULL,
+    event_signature  VARCHAR NOT NULL,
+    log_index        INTEGER NOT NULL
+);
+
+/**
+ * BRIDGING DATA
+ */
+
+CREATE TABLE IF NOT EXISTS deposits (
+	guid                 VARCHAR PRIMARY KEY NOT NULL,
+
+    -- Event causing the deposit
+    initiated_l1_event_guid VARCHAR NOT NULL REFERENCES l1_contract_events(guid),
+
+    -- Deposit information (do we need indexes on from/to?)
+	from_address     VARCHAR NOT NULL,
+	to_address       VARCHAR NOT NULL,
+	l1_token_address VARCHAR NOT NULL,
+	l2_token_address VARCHAR NOT NULL,
+	amount           NUMERIC NOT NULL,
+	data             BYTEA NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS withdrawals (
```

Comment:
For this table we are going to have to have a way of also accounting for withdrawals that failed.   When they fail the tx is kinda "canceled" and the user has to restart from scratch.    They can fail from not providing enough gas.   We have a hack in the SDK that makes it possible to estimate the correct amount of gas but anybody not using the sdk could have relays fail because of this